### PR TITLE
Change this to a monorepo

### LINKED ISSUE #12: Convert this to a monorepo
#### Description
Depending on how #11 discussion goes, we are going to want to turn this into a monorepo.

- [x] rename this repository to `appliances`
- [x] enable yarn packages
- [x] move the appliance-core code to a `packages/core` directory
- [ ] document publication practices

Based on what we learned in the base monorepo, I'm not sure we want or need [lerna](https://www.npmjs.com/package/lerna) for this first implementation.

#### Relevant Resources / Research

- We already went through some of this before over in the [base](github.com/tvkitchen/base) repo.

#### Related Issues
- Issue #11

### PR DESCRIPTION
**Pull request by slifty**
## Description
This PR adds monorepo functionality to this repository, using yarn workspaces.

This PR moves the previous `appliance-core` code to a new `core` package within the monorepo

## Due Diligence Checklist
- [x] Tests
- [x] Documentation
- [x] Consolidated commits
- [x] Relevant labels assigned

## Steps to Test
1. `yarn test`

## Deploy Notes
- `yarn install`

## Related Issues
Resolves #12

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,23 @@
-# TV Kitchen Appliance Core
+[![Build Status](https://travis-ci.com/tvkitchen/base.svg?branch=master)](https://travis-ci.com/tvkitchen/base)
```
Comment (reefdog) [inline README.md:1]: Whoops, copypasta! Needs to point to `tvkitchen/appliances`, not `tvkitchen/base`.

File: packages/core/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# TV Kitchen Appliance Core
+
+This repository contains core classes and constants for use by TV Kitchen Appliances and code that interacts with TV Kitchen Appliances.
```
Comment (reefdog) [inline packages/core/README.md:3]: Now that this is a monorepo and this README specific to the core appliance, the language here (and below) should change from "This repository" to "This package", shouldn't it?

```suggestion
This package contains core classes and constants for use by TV Kitchen Appliances and code that interacts with TV Kitchen Appliances.
```

File: packages/core/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Key Repository Contents
+
+This repository has some supporting classes, but the content that will be relevant to others are:
```
Comment (reefdog) [inline packages/core/README.md:7]:
```suggestion
This package has some supporting classes, but the content that will be relevant to others are:
```

File: packages/core/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# TV Kitchen Appliance Core
+
+This repository contains core classes and constants for use by TV Kitchen Appliances and code that interacts with TV Kitchen Appliances.
+
+## Key Repository Contents
```
Comment (reefdog) [inline packages/core/README.md:5]:
```suggestion
## Key Package Contents
```

File: packages/core/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+* `AbstractAppliance` is a class which all TV Kitchen Appliances should extend and implement.
+
+## Implementing an Abstract Appliance
+
+An implemented TV Kitchen Appliance must override the following methods which are outlined in the [IAppliance interface](https://github.com/tvkitchen/base/blob/master/packages/interfaces/src/IAppliance.js):
+
+1. `getInputTypes`: returns an array of strings that represent the data types that the appliance accepts.
+2. `getOutputTypes`: returns an array of strings that represent the data types that the appliance produces.
+3. `isValidPayload`: returns a boolean indicating if a given `Payload` (instance of data) meets the conditions of the appliance.
+4. `invoke`: will actually process the data that has been collected by the appliance so far.
```
Comment (reefdog) [inline packages/core/README.md:18]: There are a few more methods an appliance will need to define, so I think this list is out of date, but I think that's out of scope of this PR and will open a new issue. I'm just noting it to blunt the surprise when that issue comes in.

File: packages/core/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+## About the TV Kitchen
+
+TV Kitchen is a project of [Bad Idea Factory](https://biffud.com).  Learn more at [the TV Kitchen project site](https://tv.kitchen).
```
Comment (reefdog) [inline packages/core/README.md:22]: This should really be "Learn less at" tbh jkjk

File: packages/core/README.md (hunk 5 above)
Comment (slifty) [reply packages/core/README.md:18]: I believe those are the only ones! (for now -- see [this impending doom discussion](https://github.com/tvkitchen/appliances/issues/19))

The rest of the IAppliance methods are implemented in AbstractAppliance.

Comment (reefdog) [reply packages/core/README.md:18]: Oh, right right right. Sorry, I followed the link to the `IAppliance` and was like "wait there are more methods that throw `NotImplementedError`! But you're right, those three are (for now) defined by `AbstractAppliance` directly.

Sorry sorry resolved!

File: README.md [hunk 7]
```diff
@@ -1,23 +1,23 @@
-# TV Kitchen Appliance Core
+[![Build Status](https://travis-ci.com/tvkitchen/appliances.svg?branch=master)](https://travis-ci.com/tvkitchen/base)
```
Comment (reefdog) [inline README.md:1]: You changed one of them but not the other!

Comment (slifty) [reply README.md:1]: Oh.............. oh no.

HOTFIX TIME!

File: package.json [hunk 8]
```diff
@@ -1,19 +1,26 @@
 {
-  "name": "@tv-kitchen/appliance-template",
+  "name": "root",
+  "private": true,
   "version": "1.0.0",
   "description": "A template for TV Kitchen Appliances.",
```
Comment (reefdog) [inline package.json:5]: Whoops, I suppose we should've updated this description, too. 😬

Comment (slifty) [reply package.json:5]: ............. hotfix time...........

**Pull request merged**